.ado/publish.yml

```diff
@@ -45,13 +54,8 @@ jobs:
           script: node scripts/bump-oss-version.js --nightly
         condition: eq(variables['Build.SourceBranch'], 'refs/heads/master')
 
-      - script: npm publish --tag nightly --registry https://registry.npmjs.org/ --//registry.npmjs.org/:_authToken=$(npmAuthToken)
-        displayName: Publish nightly react-native-macos to npmjs.org
-        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
-
-      - script: npm publish --tag $(Build.SourceBranchName) --registry https://registry.npmjs.org/ --//registry.npmjs.org/:_authToken=$(npmAuthToken)
-        displayName: Publish stable react-native-macos to npmjs.org
-        condition: and(succeeded(), ne(variables['Build.SourceBranch'], 'refs/heads/master'))
+      - script: npm publish --tag ${{npmDistTag}} --registry https://registry.npmjs.org/ --//registry.npmjs.org/:_authToken=$(npmAuthToken)
+        displayName: Publish ${{npmDistTag}} react-native-macos to npmjs.org
```

Comment:
The docs say that compile time expressions can be used anywhere, so I believe this should work.

Reply:
I think it should. The only downside with compile time expressions is that it isn't always well defined what is available at compile time. E.g. we've had bugs in RNW CI before where a builtin variable gets redefined and the compile-time version is something wrong.

Reply:
It would be great if there were a UI that would let you do a dry-run, but couldn’t find anything of the sorts 🤷‍♂️